editor/components/post-taxonomies/check.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External Dependencies
+ */
+import { connect } from 'react-redux';
+import { filter, includes } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { withAPIData } from '@wordpress/components';
+import { compose } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentPostType } from '../../store/selectors';
+
+export function PostTaxonomiesCheck( { postType, taxonomies, children } ) {
+	const availableTaxonomies = filter( taxonomies.data, ( taxonomy ) => includes( taxonomy.types, postType ) );
+	if ( availableTaxonomies.length === 0 ) {
+		return null;
+	}
+
+	return children;
+}
+
+const applyConnect = connect(
+	( state ) => {
+		return {
+			postType: getCurrentPostType( state ),
+		};
+	},
+);
+
+const applyWithAPIData = withAPIData( () => ( {
+	taxonomies: '/wp/v2/taxonomies?context=edit',
```

Comment:
In file /editor/components/post-taxonomies/index.js with are repeating the applyConnect and applyWithAPIData logic. An option to avoid this repetition would be moving the grab taxonomies logic (applyConnect and applyWithAPIData) to file sidebar/post-taxonomies/index.js  and pass the available taxonomies to PostTaxonomiesForm and to PostTaxonomiesCheck (or do the check directly in sidebar/post-taxonomies/index.js). Regarding the checking instead of filter + checking array length 0, we can use "some" so we avoid iterating over all the array and building a new filtered array.

Reply:
Funny enough, I commented on your PR before replying here :) #4175 

So, the idea is that we do not guarantee that someone using the `PostTaxonomies` is wrapping it with the `Check` since it's a reusable component. So it's fine to duplicate to ensure components portability.

Reply:
Hi @youknowriad, I added this comment before checking your review. I understood the reasons of the duplication, so the only minor change I would do before the merge would be changing PostTaxonomiesCheck to something like `const hasTaxonomies = some( taxonomies.data, ( taxonomy ) => includes( taxonomy.types, postType ) ); if( hasTaxonomies )...`. But the performance difference should not be noticeable in this case so feel free to ignore this nitpick.

Reply:
I like this suggestion, updating.